bin/adhoc-scripts/createPileupObjects.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+This script can be used to parse the WMCore campaign documents
+and extract the equivalent pileup configuration.
+
+Attributes that are used from the campaign documents are:
+ * CampaignName: to be defined under the pileup "campaignName" attr
+ * Secondaries: to fetch the pileup dataset name, type and expected location
+
+where the pileup document minimal schema is:
+{
+    'pileupName': "",
+    'pileupType': "",
+    'expectedRSEs': [],
+    'campaigns': [],
+    'active': False
+}
+
+Example usage is:
+==> Construct pileup documents from the production campaigns
+python3 createPileupObjects.py --url=https://cmsweb.cern.ch --fout=alan.json
+
+==> Inject the previous pileup doc dump into your test cluster
+python3 createPileupObjects.py --url=https://cmsweb-test9.cern.ch --fin=alan.json --inject
+"""
+import argparse
+import json
+import logging
+import os
+import sys
+from pprint import pprint
+
+try:
+    from WMCore.Services.pycurl_manager import RequestHandler
+except ImportError:
+    print("WMCore/RequestHandler environment not set. Faking it")
+    RequestHandler = None
+try:
+    from WMCore.Services.ReqMgrAux.ReqMgrAux import ReqMgrAux
+except ImportError:
+    print("WMCore/ReqMgrAux environment not found. Faking it")
+    ReqMgrAux = None
+
+
+class OptionParser():
+    """Class to parse the command line arguments"""
+
+    def __init__(self):
+        "User based option parser"
+        self.parser = argparse.ArgumentParser(prog='PROG')
+        self.parser.add_argument("--fin", action="store",
+                                 dest="fin", default="", help="Input file")
+        self.parser.add_argument("--fout", action="store",
+                                 dest="fout", default="",
+                                 help="Output json file with all the pileup configurations created")
+        self.parser.add_argument("--verbose", action="store",
+                                 dest="verbose", default=0, help="verbosity level")
+        self.parser.add_argument("--url", action="store",
+                                 dest="url", default="https://cmsweb-testbed.cern.ch",
+                                 help="Server url to fetch and create pileup configurations")
+        self.parser.add_argument("--inject", action="store_true",
+                                 dest="inject", default=False,
+                                 help="Inject pileup documents for real")
+
+
+def getPUSchema(pileupName, pileupDocs, logger):
+    """
+    Get a minimalistic pileup data schema, or return an
+    existent pileup object.
+
+    :param pileupName: string with the pileup name
+    :param pileupDocs: list of pileup documents
+    :return: a dictionary with the pileup configuration
+    """
+    for doc in pileupDocs:
+        if pileupName == doc["pileupName"]:
```

Comment:
this code assumes that doc contains pileupName key which may not be the case. Please add protection over here.